src/software/simulated_tests/non_terminating_validation_functions/ball_always_moves_forward_validation.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#include <gtest/gtest.h>
+
+#include "software/geom/algorithms/contains.h"
+#include "software/simulated_tests/validation/validation_function.h"
+#include "software/world/world.h"
+
+/**
+ * Checks that the ball always moves forward until it is scored.
+ *
```

Comment:
nit: the validation function doesn't actually check for a goal, just that it never moves backward, can we omit the "until it is scored"? I think this would be okay since the validation function allows for a x-position increase of 0, so if a goal is scored the position would just not increase further.

Reply:
I'm also thinking, could this be named something like `ball_never_moves_backward` instead, since the validation function checks if the current position is _less_ than the previous position, not that it is _less_ than or _equal_ to the previous position?

Reply:
you're right tara! fixed it